src/VecSim/containers/data_blocks_container.cpp

```diff
@@ -0,0 +1,80 @@
+#include "data_blocks_container.h"
+
+DataBlocksContainer::DataBlocksContainer(size_t blockSize, size_t elementBytesCount,
+                                         std::shared_ptr<VecSimAllocator> allocator,
+                                         unsigned char _alignment)
+    : VecsimBaseObject(allocator), RawDataContainer(), element_bytes_count(elementBytesCount),
+      element_count(0), blocks(allocator), block_size(blockSize), alignment(_alignment) {}
+
+DataBlocksContainer::~DataBlocksContainer() = default;
+
+size_t DataBlocksContainer::size() const { return element_count; }
+
+size_t DataBlocksContainer::blockSize() const { return block_size; }
+
+size_t DataBlocksContainer::elementByteCount() const { return element_bytes_count; }
+
+RawDataContainer::Status DataBlocksContainer::addElement(const void *element, size_t id) {
+    assert(id == element_count); // we can only append new elements
+    if (element_count % block_size == 0) {
+        blocks.emplace_back(this->block_size, this->element_bytes_count, this->allocator,
+                            this->alignment);
+    }
+    blocks.back().addElement(element);
+    element_count++;
+    return OK;
+}
```

Comment:
Please use explicit scoping
```suggestion
    return Status::OK;
```
A child class might include an enum with the same enumaration (`OK`)
The easiest way to identify all the locations that require explicit scoping is to add 
`enum Status2 { OK = 0, ID_ALREADY_EXIST, ID_NOT_EXIST, ERR };` to `DataBlocksContainer` and see where the comilation fails